design-proposals/emts-scale-provisioning.md

```diff
@@ -0,0 +1,255 @@
+   The detailed design for the deployment time configuration and profile settings for both EMF and EIM will be addressed in a separate design document.
+
+### Legacy PXE provisioning workflow
+
+The solution relies on the capabilities provided by Tinkerbell SMEE. SMEE provides a minimal DHCP and TFTP server implementations that support PXE boot.
+The DHCP server provides necessary PXE info to machines initiating the PXE boot, while the TFTP server provides the iPXE binary that can be chain-loaded at boot time by PXE.
+The iPXE binary provided by the SMEE's TFTP server is a third-party script, maintained by the Tinkerbell project.
+
+The solution assumes that the local EIM-provided DHCP/TFTP server is only used to initiate boot via legacy PXE (as legacy PXE has limited capabilities, i.e., it doesn't support HTTPS boot),
+but, once chain-loaded to iPXE, the Micro-OS is downloaded and used to drive provisioning.
+In other words, the only difference between the new PXE-based boot and HTTP-based boot is how the OS provisioning is triggered. The subsequent workflow remains the same -
+it leverages Micro-OS, device discovery and Tinkerbell workflow to complete OS provisioning.
+
+**NOTE1**: Customers should provide their own local DHCP server for dynamic IP address assignment.
+
+**NOTE2**: The workflow assumes that the EIM Standalone is deployed locally on customers' premises and ENs have direct connectivity with EIM services.
+
+The high-level PXE-based provisioning workflow is as follows:
+
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+
+  box LightYellow Edge Node
+  participant bios as BIOS (PXE)
+  participant ipxe as iPXE
+  participant microos as Micro-OS
+  end
+
+  box rgb(235,255,255) Standalone Edge Infrastructure Manager, on-prem 
+  participant smee as Tinkerbell SMEE
+  participant pa as Provisioning Nginx
+  participant inv as Inventory / API
+  end
+
+  participant user as User
+
+  user->>inv: Pre-register ENs with SN/UUID
+  note over bios,ipxe: PXE boot starts
+
+  bios->>smee: DHCP Discover
+  smee->>bios: DHCP reply with tftp://<smee-ip>/ipxe.efi
+
+  bios->>+smee: Downloads ipxe.efi via TFTP
+  smee->>-bios: [ipxe.efi]
+
+  bios->>ipxe: Leaves PXE context, taken over by iPXE
+
+  ipxe->>smee: DHCP Discover
+  smee->>ipxe: DHCP reply with http://<nginx-ip>/tink-stack/boot.ipxe
+
+  ipxe->>+pa: Download boot.ipxe
+  pa->>-ipxe: [boot.ipxe]
+
+  ipxe->>ipxe: Chainloads to boot.ipxe
+
+  ipxe->>+pa: Download Micro-OS
+  pa->>-ipxe: [Micro-OS image]
+
+  note over ipxe,microos: Boots into Micro-OS
+
+  note over microos,inv: OS provisioning continues with the standard flow, through Micro-OS and Tinkerbell workflow
+```
+
+1. Users perform standard EN registration via UI or Bulk Import Tool. For each EN we can define OS profile and additonal settings (e.g., site, local account).
```

Comment:
> Users perform standard EN registration via UI

The registration flow is not the standard one as whether the Cluster is installed or not entirely depend on the selection of the OS profile (via cloud-init). We should make sure that the user only has the `<project_id>_ir/irw` roles so that the UI can hide the Cluster creation steps from the registration/provisioning flow.

cc @satya-in @soniabha-intc